warp10/src/main/java/io/warp10/continuum/gts/GTSEncoder.java

```diff
@@ -405,7 +417,7 @@ public synchronized int addValue(long timestamp, long location, long elevation,
       // Strip trailing zero so we optimize the representation
       doubleValue = doubleValue.stripTrailingZeros();
 
-      if (null != lastBDValue && 0 == lastBDValue.compareTo(doubleValue)) {
+      if (validLastBDValue && null != lastBDValue && 0 == lastBDValue.compareTo(doubleValue)) {
```

Comment:
Ditto, only testing `validLastBDValue` should be enough to be sure `lastBDValue` is non-null. It can be argued that the nullity check prevents a NPE though.